### PR TITLE
XAWS-5: Added CDK code for configuring production XWiki with MySQL in AWS account

### PR DESCRIPTION
**Pull request by sanchita141011**
Added code to configure stacks for provisioning VPC with four subnets, RDS using AWS Aurora, File-System using AWS EFS, and Application Loadbalancer. Details of the Architecture is explained [here](https://design.xwiki.org/xwiki/bin/view/Proposal/Elastic%20Compute%20Cloud-based%20Amazon%20Machine%20Image%20for%20XWiki/). Also added an easy-to-edit config file to change the XWiki version, AWS region, and account number.

### CODE REVIEW

File: xwiki-production-cdk/Bin/xwiki-production-cdk.ts [hunk 1]
```diff
@@ -17,21 +18,26 @@
...
-export class XwikiProductionCdkStack extends cdk.Stack {
-  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
-    super(scope, id, props);
+const app = new cdk.App()
 
-    const queue = new sqs.Queue(this, 'XwikiProductionCdkQueue', {
-      visibilityTimeout: cdk.Duration.seconds(300)
-    });
+const env = {
+  account: accountNumber,
```
Comment (sachin10101998) [inline xwiki-production-cdk/Bin/xwiki-production-cdk.ts]: Is there a way to move awy from this hard coded Account number?

File: xwiki-production-cdk/lib/config.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const region = 'eu-central-1'; // region in which you want to configure xwiki instance
+export const xwikiVesrion = 'xwiki:mysql-tomcat' //or 'xwiki:mysql-stable-tomcat'
```
Comment (sachin10101998) [inline xwiki-production-cdk/lib/config.ts]: Is this the latest version?

File: xwiki-production-cdk/lib/config.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const region = 'eu-central-1'; // region in which you want to configure xwiki instance
```
Comment (sachin10101998) [inline xwiki-production-cdk/lib/config.ts]: Get it rom default region of aws credentials file . if no default region is present, use "us-east-1"

File: xwiki-production-cdk/lib/config.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+export const xwikiVesrion = 'xwiki:mysql-tomcat' //or 'xwiki:mysql-stable-tomcat'
+export const accountNumber= 'XXXXXXXXXXXX' //account number of your AWS account
```
Comment (sachin10101998) [inline xwiki-production-cdk/lib/config.ts]: Move away from this similar to Demo one.

File: xwiki-production-cdk/lib/stacks/xwiki-stacks.ts [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+import { Secret } from '@aws-cdk/aws-secretsmanager'
+import { ApplicationLoadBalancer, ApplicationProtocol } from '@aws-cdk/aws-elasticloadbalancingv2'
+import { xwikistackprops } from '../models/xwiki-stack-models'
+import { xwikiVesrion } from '../config'
+
+export class XwikiProductionStacks extends Stack {
+    constructor (scope: App, id: string, props: xwikistackprops) {
+      super(scope, id, props)
+
+      const xwikiEncryptionKey = new Key(this, 'XWikiEncryptionKey', {
```
Comment (sachin10101998) [inline xwiki-production-cdk/lib/stacks/xwiki-stacks.ts]: Add comments everywhere regarding the use of each construct and why it's needed.

File: xwiki-production-cdk/lib/stacks/xwiki-stacks.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+          'DB_HOST': xwikiRds.clusterEndpoint.hostname,
+          'DB_DATABASE': 'xwiki',
+          'DB_USER': 'xwikimysql'
+        },
+        logging: new AwsLogDriver({
+          logGroup: xwikiLogGroup,
+          streamPrefix: `xwiki`
+        }),
+        secrets: {
+          'DB_PASSWORD': EcsSecret.fromSecretsManager(xwikiRdsPwSecret)
```
Comment (sachin10101998) [inline xwiki-production-cdk/lib/stacks/xwiki-stacks.ts:165]: How will the end user set this password?

File: xwiki-production-cdk/package.json [hunk 7]
```diff
@@ -11,19 +11,33 @@
...
     "ts-node": "^9.0.0",
     "typescript": "~3.9.7"
   },
   "dependencies": {
-    "@aws-cdk/aws-sns": "1.95.0",
-    "@aws-cdk/aws-sns-subscriptions": "1.95.0",
-    "@aws-cdk/aws-sqs": "1.95.0",
-    "@aws-cdk/core": "1.95.0"
+    "@aws-cdk/aws-ec2": "1.105.0",
+    "@aws-cdk/aws-ecs": "1.105.0",
```
Comment (sachin10101998) [inline xwiki-production-cdk/package.json:32]: Upgrade to monocdk. It's better and we don't need to manage versions of each module.

**Review by sachin10101998 — COMMENTED** [review]
Added few comments. Please add relevant comments everywhere and address all comments.

File: xwiki-production-cdk/Bin/xwiki-production-cdk.ts (hunk 1 above)
Comment (sanchita141011) [reply xwiki-production-cdk/Bin/xwiki-production-cdk.ts]: Done

File: xwiki-production-cdk/lib/config.ts (hunk 3 above)
Comment (sanchita141011) [reply xwiki-production-cdk/lib/config.ts]: Changed to us-east-1

File: xwiki-production-cdk/lib/config.ts (hunk 4 above)
Comment (sanchita141011) [reply xwiki-production-cdk/lib/config.ts]: Done

File: xwiki-production-cdk/lib/stacks/xwiki-stacks.ts (hunk 6 above)
Comment (sanchita141011) [reply xwiki-production-cdk/lib/stacks/xwiki-stacks.ts:165]: This is the auto-generated password of length 16 without punctuations. If the end-user wants to change it, he can do so using AWS console

File: xwiki-production-cdk/lib/stacks/xwiki-stacks.ts (hunk 5 above)
Comment (sanchita141011) [reply xwiki-production-cdk/lib/stacks/xwiki-stacks.ts]: Done. Please suggest changes if required :)

File: xwiki-production-cdk/package.json (hunk 7 above)
Comment (sanchita141011) [reply xwiki-production-cdk/package.json:32]: Will upgrade to monocdk together for both production and demo installation.

**Pull request merged**